content/2022/prepoznavanje-govora.md

```diff
@@ -2,3 +2,239 @@
+
+$$
+\min _w \lambda\|w\|^2+\sum_{i=1}^n\left(1-y_i\left\langle x_i, w\right\rangle\right)_{+}
+$$
+
+Nakon toga možemo izvesti gradijente za ažuriranje vrednosti težina modela:
+
+$\frac{\delta}{\delta w_k} \lambda\|w\|^2=2 \lambda w_k \\$
+
+$\frac{\delta}{\delta w_k}\left(1-y_i\left\langle x_i, w\right\rangle\right)_{+}= \begin{cases}0, & \text { if } y_i\left\langle x_i, w\right\rangle \geq 1 \\ -y_i x_{i k}, & \text { else }\end{cases}$
+
+Težine ažuriramo zavisno od toga da li je naš klasifikator tačno klasifikovao novi podatak ili ne. Ukoliko jeste, ažuriramo samo gradijent regularizacionog parametra:
+
+$$
+w=w-\alpha \cdot(2 \lambda w)
+$$
+
+U suprotnom, ako je model napravio grešku, moramo da uključimo i funkciju gubitka u račun:
+
+$$
+w=w+\alpha \cdot\left(y_i \cdot x_i-2 \lambda w\right)
+$$
+
+##### 6. Konvolucione neuronske mreže
+
+Metoda konvolucionih neuronskih mreža pomaže za klasifikaciju podataka pomoću tehnike dubokog učenja. Neuronske mreže rade po principu čovečjeg mozga (odatle i naziv): dobija određene podatke koji se obično nalaze u formatu baze podataka, obrađuje ih i vraća rezultate. Kontrolom rezultata obrade podataka se ta mreža trenira. Ona uči na svojim greškama i poboljšava rezultate obrade.
+
+Konvoluciona neuronska mreža korišćena u ovom projektu sastoji se iz 5 konvolucionih slojeva, koristi se 4 slojeva sažimanja, kao i 3 potpuno povezana sloja. 
+
+Ceo proces može se svesti na sledeće korake: 
+- Spektrogram se prvo obrađuje konvolucijom i ReLU-om
+- Smanjujemo veličinu obrađene slike pooling slojem
+- Ponavljamo ovaj proces
+
+Konvolucija (po čemu nastaje termin konvolucione neuronske mreže) u obradi slike je operator koji predstavlja sužavanje početne slike množenjem iste određenim filterom. 
+
+Konvolucija kao bitne detalje posmatra one koji su mnogo puta uhvaćeni u kernelu. Problem može da se desi kada kernel ne zahvata ivice dosta puta, te može mnogo da smanji određenu sliku, a samim tim i da se reši ivičnih detalja. Ako do te pojave dođe, koristi se tehnika koja se zove padding. 
+
+Padding označava dodavanje piksela na ivice. Samim tim, kada konvolucija radi svoj posao, ona će svojim kernelom mnogo puta pokriti tu površinu. 
+
+ReLU (rectified linear activation function / rectified linear unit) je funkcija koja negativnim vrednostima daje nulu, a pozitivne ostavlja kakve jesu. Time dobijamo nelinearan model.
+
+![Funkcija](static\images\fja.png)
+
+Kroz neuronsku mrežu se propušta već napravljen spektrogram, kao i labele tih spektrograma koje mreža treba da prepozna.
+
+Za treniranje mreže koriste se dve metode simultano: loss funkcija i back propagation.
+
+Cost funkcija (funkcija troškova ili gubitka) jeste funkcionalna veza željenog outputa i dobijenog outputa u funkciji, a loss funkcija je srednja vrednost svih cost funkcija.
```

Comment:
Kriterijumska funkcija, ne funkcija troškova ili gubitka. Takođe, kriterijumska funkcija je usrednjena vrednost svih funkcija greške (loss function), nije obrnuto kao što ste vi napisali.

Reply:
Rešeno

Reply:
Rešeno